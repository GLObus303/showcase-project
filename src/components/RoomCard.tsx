import { Link } from 'react-router-dom';

import { RoomType } from '../model/Room';
import { Rating } from './Rating';

type RoomCardProps = Pick<
  RoomType,
  'imageUrl' | 'name' | 'rating' | 'subtitle' | 'id'
>;

export const RoomCard: React.FC<RoomCardProps> = ({
  imageUrl,
  name,
  rating,
  subtitle,
  id,
}) => (
  <div className="flex flex-col border border-black rounded-md cursor-pointer hover:shadow-2xl transition-shadow">
    <Link to={`room/${id}`}>
      <img src={imageUrl} alt="Room" className="w-full" />

      <div className="p-2 flex flex-col">
        <div className="text-lg md:text-2xl font-semibold">{name}</div>
        <div className="pb-4 md:text-lg">{subtitle}</div>
        <div className="flex justify-end items-center">
          <Rating rating={rating} />
        </div>
      </div>
    </Link>
  </div>
);
