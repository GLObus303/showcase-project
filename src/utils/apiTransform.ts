import {
  DocumentData,
  DocumentSnapshot,
  QuerySnapshot,
} from 'firebase/firestore';

export const transformSnapshotToList = <T>(
  snapshot: QuerySnapshot<DocumentData>,
): T[] => {
  const data: any[] = [];

  snapshot?.forEach((doc) => data.push({ id: doc.id, ...doc.data() }));

  return data;
};

export const transformSnapshotToData = <T>(
  document: DocumentSnapshot<DocumentData>,
) => ({ id: document.id, ...document.data() } as unknown as T);
