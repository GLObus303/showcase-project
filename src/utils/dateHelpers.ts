import { format } from 'date-fns';
import { Timestamp } from 'firebase/firestore';

import { FirebaseDate } from '../model/Review';

export const fireBaseTimestampToDate = (fireBaseDate: FirebaseDate) =>
  new Timestamp(fireBaseDate.seconds, fireBaseDate.nanoseconds).toDate();

export const formatFirebaseDate = (fireBaseDate: FirebaseDate) =>
  format(fireBaseTimestampToDate(fireBaseDate), 'dd.MM.yyyy');
