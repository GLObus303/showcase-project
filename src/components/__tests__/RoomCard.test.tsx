import { screen } from '@testing-library/react';

import { renderWithContext } from '../../test/utils/renderWithContext';
import { RoomCard } from '../RoomCard';
import { roomMock } from '../../test/mocks/roomMock';

describe('RoomCard', () => {
  it('should render header, logo redirects to homepage', () => {
    const { container } = renderWithContext(<RoomCard {...roomMock} />);

    expect(screen.getByText(/namemock/i)).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
