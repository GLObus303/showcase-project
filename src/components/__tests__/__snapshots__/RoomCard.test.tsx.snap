// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomCard should render header, logo redirects to homepage 1`] = `
<div>
  <div
    class="flex flex-col border border-black rounded-md cursor-pointer hover:shadow-2xl transition-shadow"
  >
    <a
      href="/room/idMock"
    >
      <img
        alt="Room"
        class="w-full"
        src="imageUrlMock"
      />
      <div
        class="p-2 flex flex-col"
      >
        <div
          class="text-lg md:text-2xl font-semibold"
        >
          nameMock
        </div>
        <div
          class="pb-4 md:text-lg"
        >
          subtitleMock
        </div>
        <div
          class="flex justify-end items-center"
        >
          <div
            class="flex items-center"
          >
            <div
              class="text-lg md:text-xl"
            >
              3.5
            </div>
            <svg
              class="ml-1 w-5 h-5 mb-1 text-ant"
            >
              star.svg
            </svg>
          </div>
        </div>
      </div>
    </a>
  </div>
</div>
`;
